lib/storage/file.js

```diff
@@ -1070,18 +1124,24 @@ File.prototype.startResumableUpload_ = function(stream, metadata) {
 File.prototype.startSimpleUpload_ = function(stream, metadata) {
   var that = this;
 
+  var reqOpts = {
+    qs: {
+      name: that.name
+    },
+    uri: util.format('{base}/{bucket}/o', {
+      base: STORAGE_UPLOAD_BASE_URL,
+      bucket: that.bucket.name
+    })
+  };
+
+  if (this.explicitGeneration) {
+    reqOpts.qs.ifGenerationMatch = this.explicitGeneration;
```

Comment:
I think that's what's happening. The API request is still made, but if the current generation isn't the one the user gave, the API call should fail.